src/main/java/legalthings/lto_api/utils/main/CryptoUtil.java

```diff
@@ -0,0 +1,138 @@
+package legalthings.lto_api.utils.main;
+
+import org.abstractj.kalium.crypto.Box;
+import org.abstractj.kalium.crypto.Hash;
+import org.abstractj.kalium.crypto.Random;
+import org.abstractj.kalium.crypto.Util;
+
+
+import org.abstractj.kalium.keys.VerifyKey;
+
+import static org.abstractj.kalium.NaCl.*;
+import static org.abstractj.kalium.crypto.Util.zeros;
+
+import legalthings.lto_api.utils.core.JsonObject;
+//import org.abstractj.kalium.crypto.Hash;
+
+import org.libsodium.jni.Sodium;
+import org.libsodium.jni.NaCl;
+
+import jnr.ffi.LibraryLoader;
+
+public class CryptoUtil {
+	public static interface Sodium {
+		int crypto_sign_bytes();
+		int crypto_sign_publickeybytes();
+		int crypto_sign_secretkeybytes();
+		int crypto_box_publickeybytes();
+		int crypto_box_secretkeybytes();
+		int crypto_box_noncebytes();
+		int crypto_generichash_bytes();
+		int crypto_scalarmult_curve25519_bytes();
+		
+		int crypto_sign_detached(byte[] sig, long siglen[], byte[] m, long mlen, byte[] sk);
+		int crypto_sign_verify_detached(byte[] sig, byte[] m, long mlen, byte[] pk);
+		int crypto_generichash(byte[] out, int outlen, byte[] in, int inlen, byte[] key, int keylen);
+		int crypto_sign_ed25519_pk_to_curve25519(byte[] curve25519_pk, byte[] ed25519_pk);
+		int crypto_sign_ed25519_sk_to_curve25519(byte[] curve25519_pk, byte[] ed25519_pk);
+		int crypto_sign_seed_keypair(byte[] pk, byte[] sk, byte[] seed);
+		int crypto_box_seed_keypair(byte[] pk, byte[] sk, byte[] seed);
+    }
+	
+	private static Sodium sodium = null;
+	
+	static {
+		sodium = LibraryLoader.create(Sodium.class).load("/usr/local/lib/libsodium.so");
```

Comment:
This will only make in work on OS X environments. Please use the loader instance provided by the library.